### ISSUE TITLE
Json output is an empty list "[ ]"

### ISSUE THREAD

**Issue by visbyone** [thtrieu/darktf#932]
Hi,
I'm training my own dataset which has 10 classes. I've made the needed cfg changes (new copy, classes and filters), and it stagnates with loss around 4, which results in an empty json file. Based on #80 decided to take a step back and do a overfitting training using 3 images and 4 classes. I'm using this command:

(classes 10, filters 75)
`python flow --train --model cfg/tiny-yolo-voc-10c.cfg --dataset Dataset/ --annotation Anno/ --batch 16 --lr 1e-5 --epoch 500 --save 200`

And this for overfitting (classes 4, filters 45)
`python flow --train --model cfg/tiny-yolo-voc-4cc.cfg --dataset Data/ --annotation Annotations/  --epoch 100 --lr 1e-5`

I tested several learning rate values (in both normal and overfitting training), but any of them reach a loss inferior to 1.

I also tried to change other cfg values. My images sizes are 228x96, which are smaller than the usual. I tried to use some darknet suggestions to detect small objects by changing:

```
subdivisions=64
height=1088
width=1088
```
on the cfg file. But there were no improvements.

If it helps, there are no overlapping objects, so the cfg could be modified, but I'm not sure if it's possible.

Any ideas?

**Comment by anoubhav** [comment]
I have the same issue. On prediction I get an empty list. Even on the training data.

**Comment by A6a7H** [comment]
I have the same problem, I have 1 class. Tried to use tiny-yolo.cfg and yolo.cfg but returned anyway [].

**Comment by BotOreo** [comment]
I am having the same problem. Any idea how to fix this?

**Comment by jaykandakatla123** [comment]
I have same problem... can anyone help?

**Comment by ViniciusMedeirosWanderley** [comment]
I have same problem

**Comment by BotOreo** [comment]
Hey, it's been a while but I finally did it a few weeks ago. I get about 70%-90% detection or confidence rate (occasionally, but not always since there are still times where it won't detect at all). I changed how I approach about this. I am not sure if it helps you, but I did promised to share once I managed to solve my problem.

First of all, I changed and increased my datasets. Since YOLO use class probability map and combine with the bounding box confidence, it just occurred to me to use a better and clearer datasets. So I created my own datasets that I took from my own web camera and phone.

I took about 5,000 images and annotate them all (one by one, it takes a few days). Then, I divided them into 10 parts, with each parts containing around 500 images.

So I trained my model with the first part (only 500 images) and let the average loss reduced to the smallest amount possible. Then I increased the learning rate a bit (from 1x10-6 to 2x10-6, then end at 1x10-5), then let it train again to make sure it is the average loss is the smallest amount it can reached. The number of epochs really depends on how many datasets are there.

Then, I continued the training with the second part of the datasets (another 500 images). The average loss will suddenly increase a bit (maybe from 0.xx to 3.xx or 2.xx or just 1.xx, but it will increase a little bit). Trained it until I get the smallest amount possible for the average loss.

I repeated these steps until I finished training all 5,000 images. When you increase the learning rate, make sure it is not too far of a jump and sometimes it will increase for a while, before it started to decrease again. But if it keeps on increasing after a very long period, then you know that it has diverged and you need to reduce the learning rate again. So make sure you documented all the necessary checkpoint (so that you know which is good and which is isn't)

The command that I used on Windows Powershell are:

(To continue training from the latest checkpoint available)
python flow --model cfg/tiny-yolo-voc-myC.cfg --load -1 --train --annotation training/annotations --dataset training/cube_data --gpu 0.75 --epoch 1000

(To continue training from the checkpoint I wanted, which in this case, checkpoint 15300)
python flow --model cfg/tiny-yolo-voc-myC.cfg --load 15300 --train --annotation training/annotations --dataset training/cube_data --gpu 0.75 --epoch 1000


(To increase the learning rate to 1x10-5,  and train from the latest checkpoint)
python flow --model cfg/tiny-yolo-voc-myC.cfg --load -1 --train --annotation training/annotations --dataset training/cube_data --gpu 0.75 --epoch 1000 --lr 0.00001

(To increase the learning to 1x10-5, and train from the checkpoint that I wanted, which in this case 20425)
python flow --model cfg/tiny-yolo-voc-myC.cfg --load 20425 --train --annotation training/annotations --dataset training/cube_data --gpu 0.75 --epoch 1000 --lr 0.00001

lr = learning rate (I only increased it from 1x10-6 to 2x10-6 and never go beyond 1x10-5)
load = loading the checkpoint or the weights

Note : When I try my webcam inferencing, I reduce the threshold to 0.05 just to test it first, to see if there is any bounding box available even at that low of a threshold. Increase the threshold as I train my model further. Right now, I increased my threshold bit by bit just to see the improvements when I tested it as I trained my model further and set my threshold highest at 0.6 and never more.

The CFG file is edited to fit my class using 5 * (number of classes + 5) , just the same as before. The same goes with the label.

Hope I helped~ Good luck~

**Comment by mdhanna** [comment]
Was anyone else able to solve this problem?  I'm training 10 classes on 27,900 images.  Batch size 16, 100 epochs.  Trained this model for multiple days and loss got down to 7.5 before I had to kill it to free up CPU.  But I still get an empty list.
